### PR TITLE
valid odoo version from README.rst

### PR DESCRIPTION
**Pull request by kaadevelop**
valid odoo version from README.rst

### CODE REVIEW

File: pylint_odoo/checkers/itp_checks.py [hunk 1]
```diff
@@ -138,6 +143,29 @@ def _check_manifest_image(self):
             else:
                 return False
 
+    @utils.check_messages('readme-versions')
+    def _check_readme_versions(self):
+        readme_file = os.path.isfile(os.path.join(self.module_path, 'README.rst'))
+        if readme_file:
+            for line in self.readme_text:
+                urls = re.findall(
+                    'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+', line)
```
Comment (yelizariev) [inline pylint_odoo/checkers/itp_checks.py]: Add a comment whether it look for all links or for some specific. If you copy-pasted it from stackoverflow, paste a linkt too

File: pylint_odoo/checkers/itp_checks.py [hunk 2]
```diff
@@ -138,6 +143,29 @@ def _check_manifest_image(self):
             else:
                 return False
 
+    @utils.check_messages('readme-versions')
+    def _check_readme_versions(self):
+        readme_file = os.path.isfile(os.path.join(self.module_path, 'README.rst'))
+        if readme_file:
```
Comment (yelizariev) [inline pylint_odoo/checkers/itp_checks.py]: if not readme_file:
    return False

File: pylint_odoo/checkers/itp_checks.py [hunk 3]
```diff
@@ -138,6 +143,29 @@ def _check_manifest_image(self):
...
+            for line in self.readme_text:
+                urls = re.findall(
+                    'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+', line)
+                if urls != []:
+                    for url in urls:
+                        odoo_version = re.search('/(\d+.\d)/', url)
+                        if odoo_version:
+                            valid_odoo_version = self.linter._all_options[
+                                'valid_odoo_versions'].config.valid_odoo_versions[0]
```
Comment (yelizariev) [inline pylint_odoo/checkers/itp_checks.py]: You can compute variable valid_odoo_version once. No need to do it in every iteration

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: pylint_odoo/checkers/itp_checks.py [hunk 4]
```diff
@@ -138,6 +143,29 @@ def _check_manifest_image(self):
...
+    def _check_readme_versions(self):
+        readme_file = os.path.isfile(os.path.join(self.module_path, 'README.rst'))
+        if readme_file:
+            for line in self.readme_text:
+                urls = re.findall(
+                    'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+', line)
+                if urls != []:
+                    for url in urls:
+                        odoo_version = re.search('/(\d+.\d)/', url)
+                        if odoo_version:
```
Comment (yelizariev) [inline pylint_odoo/checkers/itp_checks.py]: if not odoo_version:
    continue

File: pylint_odoo/checkers/itp_checks.py (hunk 1 above)
Comment (kaadevelop) [reply pylint_odoo/checkers/itp_checks.py]: https://www.tutorialspoint.com/python/python_extract_url_from_text.htm
It look all links

File: pylint_odoo/checkers/itp_checks.py (hunk 2 above)
Comment (kaadevelop) [reply pylint_odoo/checkers/itp_checks.py]: it already is!!!

File: pylint_odoo/checkers/itp_checks.py (hunk 4 above)
Comment (kaadevelop) [reply pylint_odoo/checkers/itp_checks.py]: return False?

File: pylint_odoo/checkers/itp_checks.py (hunk 3 above)
Comment (kaadevelop) [reply pylint_odoo/checkers/itp_checks.py]: will to do!

File: pylint_odoo/checkers/itp_checks.py (hunk 1 above)
Comment (yelizariev) [reply pylint_odoo/checkers/itp_checks.py]: Ok, but add a code-comment please

File: pylint_odoo/checkers/itp_checks.py (hunk 4 above)
Comment (yelizariev) [reply pylint_odoo/checkers/itp_checks.py]: no

File: pylint_odoo/checkers/itp_checks.py (hunk 2 above)
Comment (yelizariev) [reply pylint_odoo/checkers/itp_checks.py]: not really

**Pull request merged**